RecoTracker/TrackProducer/interface/TrackProducerBase.icc

```diff
@@ -228,9 +228,9 @@ TrackProducerBase<T>::setSecondHitPattern(Trajectory* traj, T& track,
         auto const & detWithState = it->compatibleDets(tsos,*localProp,estimator);
         if(!detWithState.size()) continue;
         DetId id = detWithState.front().first->geographicalId();
-        MeasurementDetWithData measDet = measTk->idToDet(id);
-        //if(measDet->isActive() && !measDet->hasBadComponents(detWithState.front().second)){
-        if(measDet.isActive()){ 
+        MeasurementDetWithData const & measDet = measTk->idToDet(id);
+        if(measDet.isActive() && !measDet.hasBadComponents(detWithState.front().second)){
+//        if(measDet.isActive()){ 

```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant

from the context, it looks like the idea is to allow to flip one way or the other. shouldn't there be a config flag to handle this rather than a hardcoded choice?